securebanking-openbanking-uk-rs-simulator/src/main/java/com/forgerock/securebanking/openbanking/uk/rs/common/refund/FRResponseDataRefundFactory.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright © 2020 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.securebanking.openbanking.uk.rs.common.refund;
+
+import com.forgerock.securebanking.common.openbanking.uk.forgerock.datamodel.payment.*;
+
+import java.util.Optional;
+
+public class FRResponseDataRefundFactory {
```

Comment:
A re-write of https://github.com/OpenBankingToolkit/openbanking-aspsp/blob/7990cc32fc3f98238bceae67c521787501d466b2/forgerock-openbanking-uk-aspsp-common/src/main/java/com/forgerock/openbanking/common/services/openbanking/payment/ResponseRefundPaymentsFactory.java

Instead of using the numerous OB objects, I'm using equivalent FR ones (plus the `FRDomesticDataInitiation` and `FRInternationalDataInitiation` interfaces that I added to `securebanking-openbanking-uk-forgerock-datamodel`).

In general I think it's best to use FR objects internally as much as possible, as it gives us much greater flexibility (plus it makes it easier to migrate between versions). I'm trying to limit the use of OB objects to the external API only.

Reply:
Seems good to me.